plugins/connection/lxd.py

```diff
@@ -126,7 +126,7 @@ def put_file(self, in_path, out_path):
         local_cmd.extend([
             "file", "push",
             in_path,
-            "%s:%s/%s" % (self.get_option("remote"), self._host, out_path)
+            "%s:%s/%s" % (self.get_option("remote"), self.get_option("remote_addr"), out_path)
```

Comment:
Why do you only replace it here, in line 92 and in line 148?

Reply:
I speculated that ```put_file```/```fetch_file``` would trip on the same issue as the above and insert the inventory alias name instead of the actual container name (at least when using ```ansible_lxd_host```) for ```<instance>```. Using the ```copy``` module in the test playbook before my changes. I was able to test ```put_file``` and found it did insert the inventory alias name instead of the container name.